assets/sass/05-blocks/utilities/_font-sizes.scss

```diff
@@ -0,0 +1,94 @@
+// Gutenberg Font-size utility classes
+:root {
```

Comment:
Did you learn why or if :root is needed here? I'd like to learn why it can't be removed.

Reply:
It's used by Gutenberg in this way on the front end of the site.

Reply:
It's for specificity — gutenberg uses it to make sure the colors take effect over other defaults. It could be anything, like a class (`:root` has the specificity level of a class), but it's guaranteed to be on the page since it refers to the root element (`html` in a website).

Reply:
Yes. To be more specific but most likely out of scope for this PR:
The editor font sizes are the **only** styles besides the CSS variables that are placed in the :root. 
For example, the custom colors are not.

Reply:
Oh well, it is not going to matter for much longer since themes eventually will be using the _--wp--preset--_ variables.

Reply:
I see the colors are using this format `.has-black-color[class]`, which accomplishes the same thing as `:root .has-black-color`.

It's needed to override things like if the heading block has a base font size, the `:root .has-large-font-size` (or .has-large-font-size[class]`) rule will override that without needing to explicitly write out each block class.